### DISCUSSION TITLE
Help with websites

### DISCUSSION THREAD

**Discussion by hawswebmaster** [code4recovery/12-step-meeting-list#1163] [Q&A] [answered]
Hi, we have the TSML plugin and it has been wonderful except that we have areas with their own website and meeting information. If an area requests us to list their website instead of the meetings, we do. This shows up very strangely on the TSML. I have changed some of the files to list attendance option "No" to say "See Website" and "Inactive" to say "See Website" with only a little luck. I think there is more that I would need to change in the plugin files and I am not sure what that may be. Also, I am not a programmer so I am thinking there may be a code in the functions file I could add, but have no clue what that may be either. Also when the meeting shows inactive, the state is greyed out. We would like to not show the city and state when we upload these meetings as the website can represent a whole state. Such as Alabama meetings are not just in Birmingham. Please help as I would like these meetings to be reflected the way the fellowship would like them to look and have received numerous concerns with the website listings. 

The website is https://heroinanonymous.org/meetings/
Wordpress 6.2.2
PHP 7.4.33

![image](https://github.com/code4recovery/12-step-meeting-list/assets/134229710/430aa76f-b6af-41df-a261-8039f30f5650)

![image](https://github.com/code4recovery/12-step-meeting-list/assets/134229710/45c65c13-7b3c-4dbf-b6db-2f37e781d8ce)

![image](https://github.com/code4recovery/12-step-meeting-list/assets/134229710/c2427f66-920e-4f97-94c6-36d02afc324d)

**Comment by anchovie91471** [comment]
Hey @hawswebmaster! Thanks for the input!

TSML wasn't really designed to accommodate this use-case, but I'll be happy to help you get as close to your end goal as possible.

A couple of follow up questions:

1.  What changes have you made so far to get where you are (code/functions.php)?
2. Could you markup these screenshot to indicate what you'd like to see in the end?

The individual meeting page can be overridden by copying the single-meeting.php from the /templates directory in the 12 Step Meeting List plugin into your theme directory. That is where you'd remove the City/State from the meeting page.

**Reply by tim-rohrer** [reply, marked as answer]
Hi @hawswebmaster, 

This is a tricky issue. Do you know why the states (areas?) don't want to list actual meetings on the HAWS site? If they were willing, and maintained their own sites in TSML (or in some other database), you can simply add them as a JSON feed to your site. Then simply refresh from time-to-time. Both websites show the same data for an area. Easy peasy. This would work well for Arizona as they are already using TSML. This approach is the least amount of work for you in the long run.

Another option is to accept management of the area's meetings on your site, but use CSV files to import them. Or, grant the area access to your site and they can manage just those meetings.  You can still have another menu on your site for links to their websites; we consider this very important for finding the meetings in local areas.

As a consumer of meeting information, I would be confused by seeing links in the meeting finder. My opinion/experience and nothing more. I believe the best approach is whatever makes the meeting data the most accurate and easiest to access for the end users.

I suggest examining the goals for the HAWS site. If it is to have a complete listing of all HA meetings across all areas, asking local intergroups to get their meeting data into JSON feeds is probably the best; but maybe you're way past this already? :-) If that is the case, and the areas are adamantly against it, the easiest solution is just to leave them off your meeting finder and then link to their sites with a banner or another menu item.

If you still want to hack the plugin, @anchovie91471 has provided some hints to get you started. You also have the option to fork the repo and make the changes you desire (if you're a developer and comfortable with such work). However, we've found this isn't usually the best for groups because they have trouble with maintaining changes, but it can be done.

HTH,
-Tim

**Comment by wgfinley** [comment]
TSML is really meant to handle a recurring meeting listing, and that is all. If you try to do anything other than that with it, I don't think it quite fits what you're after.

Example: We have events in addition to meetings, these are not recurring. We use an Events plugin to handle those rather than trying to use TSML for them.

Similarly, you're putting links together for areas that want to maintain their own listing (you'll be very grateful for them!).  I would create a separate page for this in order to list the links to them. I see you have a meetings landing page [here](https://heroinanonymous.org/meetings1/), I'd put the link to the page for those individually listing their meetings there.

This can be something pretty simple where you just use the editor to make your list or you could even use custom posts to do it if you wanted to get fancy.

--Guy
